gcc/rust/parse/rust-parse-impl.h

```diff
@@ -1217,11 +1217,11 @@ Parser<ManagedTokenSource>::parse_outer_attribute ()
 
   if (lexer.peek_token ()->get_id () == INNER_DOC_COMMENT)
     {
-      Error error (
-	lexer.peek_token ()->get_locus (),
-	"inner doc (%<//!%> or %</*!%>) only allowed at start of item "
+      rust_error_at (
```

Comment:
This changes the behavior of the code, the error is emitted immediately instead of being collected. Sometimes in some context we wish to collect those errors and discard them. I'm not too familiar with the parser's error collection. There should be a valid reason this was done before @CohenArthur ?

Reply:
Yes, @P-E-P you are right. For `parser` & `expansion` errors, we need to use `Error` struct instead of `rust_error_at` This shouldn't be there. I need to fix this issue

- [ ] https://github.com/Rust-GCC/gccrs/issues/2385


& then update this https://github.com/Rust-GCC/gccrs/pull/2531 and this PR



- [ ]  https://github.com/Rust-GCC/gccrs/pull/2527


Thanks for reminding, I completely forgot about this issue :)